syntax = "proto3";

message Node {
 string name = 1;
  float x = 2;
  float y = 3;
   int32 parent = 4;
}


message NodeList {
 repeated Node nodes=1;
}


message MetadataItem {
 string name = 1;
  string lineage = 2;
  string date = 3;
  string country = 4;
  repeated string aa_subs = 5;
}

message MetadataList {
 repeated MetadataItem items = 1;

}


message MutationList{
  repeated int32 mutation = 1;
}

message AllNodeData {
  repeated string names = 1;
 repeated float x = 2;
 repeated float y = 3;
 repeated int32 countries = 4;
 repeated int32 lineages = 5;

 repeated MutationList mutations = 6;
 repeated int32 dates = 7;
 repeated int32 parents = 8;

}


message AllData{
  AllNodeData node_data= 1;
  repeated string country_mapping = 2;
  repeated string lineage_mapping = 3;
  repeated string mutation_mapping = 4;
  repeated string date_mapping = 5;
}
